source/Politics.cpp

```diff
@@ -206,54 +206,74 @@ bool Politics::HasDominated(const Planet *planet) const
+	// Do nothing if you have already been fined today.
+	if(fined.count(gov) || !gov->GetFineFraction())
 		return "";
+		
+	const bool isCargoScan = (scan & ShipEvent::SCAN_CARGO);
+	const bool isOutfitScan = (scan & ShipEvent::SCAN_OUTFITS);
 	
 	string reason;
 	int64_t maxFine = 0;
 	for(const shared_ptr<Ship> &ship : player.Ships())
 	{
 		// Check if the ship evades being scanned due to interference plating.
-		if(Random::Real() > 1. / (1. + ship->Attributes().Get("scan interference")))
-			continue;
+		double scanResistance = 1. / (1. + ship->Attributes().Get("scan interference"));
+		
 		if(target && target != &*ship)
 			continue;
 		if(ship->GetSystem() != player.GetSystem())
 			continue;
 		
-		if(!scan || (scan & ShipEvent::SCAN_CARGO))
+		if(!scan || isCargoScan)
 		{
-			int64_t fine = ship->Cargo().IllegalCargoFine();
-			if((fine > maxFine && maxFine >= 0) || fine < 0)
+			int64_t fine = 0;
+			if((Random::Real() <= scanResistance))
 			{
-				maxFine = fine;
-				reason = " for carrying illegal cargo.";
-
-				for(const Mission &mission : player.Missions())
+				if(!scan)
+					fine = player.Cargo().IllegalCargoFine();
+				else
+					fine = ship->Cargo().IllegalCargoFine();
+				if((fine > maxFine && maxFine >= 0) || fine < 0)
 				{
-					// Append the illegalCargoMessage from each applicable mission, if available
-					string illegalCargoMessage = mission.IllegalCargoMessage();
-					if(!illegalCargoMessage.empty())
+					maxFine = fine;
+					reason = " for carrying illegal cargo.";
+
+					for(const Mission &mission : player.Missions())
 					{
-						reason = ".\n\t";
-						reason.append(illegalCargoMessage);
+						// Append the illegalCargoMessage from each applicable mission, if available
+						string illegalCargoMessage = mission.IllegalCargoMessage();
+						if(!illegalCargoMessage.empty())
+						{
+							reason = ".\n\t";
+							reason.append(illegalCargoMessage);
+						}
+						// Fail any missions with illegal cargo and "Stealth" set
+						if(mission.IllegalCargoFine() > 0 && mission.FailIfDiscovered())
+							player.FailMission(mission);
 					}
-					// Fail any missions with illegal cargo and "Stealth" set
-					if(mission.IllegalCargoFine() > 0 && mission.FailIfDiscovered())
-						player.FailMission(mission);
 				}
 			}
 		}
-		if(!scan || (scan & ShipEvent::SCAN_OUTFITS))
+		// If you are scanned, each outfit is rolled for detection. Your cargo hold is only rolled once for detection of contraband.
+		if((!scan || isOutfitScan) && Random::Real() <= scanResistance)
 		{
+			int64_t fine = 0;
 			for(const auto &it : ship->Outfits())
 				if(it.second)
 				{
-					int64_t fine = it.first->Get("illegal");
 					if(it.first->Get("atrocity") > 0.)
 						fine = -1;
+					else
+					{
+						int numDetectedOutfits = 0;
+						fine += it.first->IllegalCargoFine();
+						for(int i = 1; i < it.second; ++i)
+							if(((!scan || isOutfitScan) && (Random::Real() <= scanResistance)))
```

Comment:
`(!scan || isOutfitScan)` will always be true here since it was the gate condition into the outfit block.